src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -1359,7 +1359,7 @@ void Terminal::AddMark(const ScrollMark& mark,
     m.end = end;
 
     // If the mark came from the user adding a mark via the UI, don't make it the active prompt mark.
-    _activeBuffer().AddMark(m, !fromUi);
+    fromUi ? _activeBuffer().AddMark(m) : _activeBuffer().StartPromptMark(m);

```

Comment:
It is very rare that using a ternary like this is the right thing to do.

Reply:
yea that felt dirty to write